src/main/java/com/toofifty/easygiantsfoundry/EasyGiantsFoundryPlugin.java

```diff
@@ -105,6 +124,23 @@ public void onGameStateChanged(GameStateChanged event)
 		}
 	}
 
+	@Subscribe
+	public void onStatChanged(StatChanged statChanged)
+	{
+		if (state.isEnabled() && state.getCurrentStage() != null)
```

Comment:
This will catch a lot of incorrect stat change calls for this purpose. If this is used it should filter out incorrect skills and only use xp drops. E.g. somebody may have boosted smithing levels that trigger this when they boost or the level falls down.